src/bridge/cockpitfsread.c

```diff
@@ -238,6 +249,19 @@ cockpit_fsread_prepare (CockpitChannel *channel)
       return;
     }
 
+  if (fstat (self->fd, &statbuf) < 0)
+    {
+      cockpit_channel_fail (channel, "internal-error",
+                            "%s: couldn't stat: %s", self->path, strerror (errno));
+      return;
+    }
+
+  if (statbuf.st_size > max_read_size)
+    {
```

Comment:
Note that this check won't work for any non-file arguments (block/char devices, pipes, sockets, and the like), st_size == 0 for these. While you are at it, would you consider adding a check for the type as well and error out on non-files?